### PR TITLE
New subnet create ux

### PR DESCRIPTION
**Pull request by felipemadero**
Closes https://github.com/ava-labs/avalanche-cli/issues/1917

### CODE REVIEW

File: cmd/subnetcmd/create.go [hunk 1]
```diff
@@ -68,31 +72,30 @@ can create a custom, user-generated genesis with a custom VM by providing
...
 	cmd.Flags().StringVar(&customVMBranch, "custom-vm-branch", "", "custom vm branch or commit")
 	cmd.Flags().StringVar(&customVMBuildScript, "custom-vm-build-script", "", "custom vm build-script")
 	cmd.Flags().BoolVar(&useRepo, "from-github-repo", false, "generate custom VM binary from github repository")
-	cmd.Flags().BoolVar(&useWarp, "warp", true, "generate a vm with warp support (needed for teleporter)")
-	cmd.Flags().BoolVar(&teleporterReady, "teleporter", false, "generate a teleporter-ready vm")
-	cmd.Flags().BoolVar(&runRelayer, "relayer", false, "run AWM relayer when deploying the vm")
+	cmd.Flags().BoolVar(&createFlags.useWarp, "warp", true, "generate a vm with warp support (needed for teleporter)")
+	cmd.Flags().BoolVar(&createFlags.useTeleporter, "teleporter", false, "interoperate with other blockchains using teleporter")
 	return cmd
 }
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply cmd/subnetcmd/create.go:100]: here https://github.com/ava-labs/avalanche-cli/pull/2048

File: pkg/vm/evmPrompts.go [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+			if err != nil {
+				return SubnetEVMGenesisParams{}, err
+			}
+		case explainOption:
+			ux.Logger.PrintToUser("The two gas fee variables that have the largest impact on performance are the gas limit, the maximum amount of gas that fits in a block, and the gas target, the expected amount of gas consumed in a rolling ten-second period.")
+			ux.Logger.PrintToUser(" ")
+			ux.Logger.PrintToUser("By increasing the gas limit, you can fit more transactions into a single block which in turn increases your max throughput. Increasing the gas target has the same effect; if the targeted amount of gas is not consumed, the dynamic fee algorithm will decrease the base fee until it reaches the minimum.")
+			ux.Logger.PrintToUser(" ")
+			ux.Logger.PrintToUser("There is a long-term risk of increasing your gas parameters. By allowing more transactions to occur on your network, the network state will increase at a faster rate, meaning infrastructure costs and requirements will increase.")
+			continue
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/vm/evmPrompts.go:384]: here https://github.com/ava-labs/avalanche-cli/pull/2050

File: cmd/subnetcmd/upgradecmd/generate.go [hunk 3]
```diff
@@ -295,6 +301,37 @@ func promptRewardManagerParams(
 	return false, nil
 }
 
+func ConfigureInitialRewardConfig() (*rewardmanager.InitialRewardConfig, error) {
+	config := &rewardmanager.InitialRewardConfig{}
+
+	burnPrompt := "Should fees be burnt?"
```
Comment (martineckardt) [inline cmd/subnetcmd/upgradecmd/generate.go:307]:
```suggestion
	burnPrompt := "By default, all transaction fees on Avalanche are burned (sent to a blackhole address). (Reward Manager Precompile)"
```

Answer options:
"I am fine with transaction fees being burned (Reward Manager Precompile OFF)"
"I want to customize the transaction fee distribution (Reward Manager Precompile ON)"

File: pkg/teleporter/teleporter.go [hunk 4]
```diff
@@ -29,12 +29,10 @@ const (
...
 
 var (
-	messengerDeployerRequiredBalance = big.NewInt(0).
-						Mul(big.NewInt(1e18), big.NewInt(10))
 	// 10 AVAX
-	TeleporterPrefundedAddressBalance = big.NewInt(0).
-						Mul(big.NewInt(1e18), big.NewInt(600))
+	messengerDeployerRequiredBalance = big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(10))
 	// 600 AVAX
+	TeleporterPrefundedAddressBalance = big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(600))
```
Comment (martineckardt) [inline pkg/teleporter/teleporter.go:35]: Rename teleporter to InterchainMessaging. Also specify what this address is for (deployer)

File: pkg/vm/allocations.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	if !ok {
+		return allocations, errors.New("unable to decode default allocation")
+	}
+
+	ux.Logger.PrintToUser("prefunding address %s with balance %s", PrefundedEwoqAddress, defaultAirdropAmount)
+	addAllocation(allocations, PrefundedEwoqAddress.String(), defaultAmount)
+	return allocations, nil
+}
+
+func addTeleporterAllocation(
```
Comment (martineckardt) [inline pkg/vm/allocations.go:56]: Rename to InterchainMessaging

**Review by martineckardt — CHANGES_REQUESTED** [review]

File: pkg/vm/evmPrompts.go [hunk 6]
```diff
@@ -0,0 +1,673 @@
...
+					}
+					params.enableNativeMinterPrecompile = true
+				}
+				break
+			}
+		case explainOption:
+			ux.Logger.PrintToUser("Every blockchain uses a token to manage access to its limited resources. For example, ETH is the native token of Ethereum, and AVAX is the native token of the Avalanche C-Chain. Users pay transaction fees with these tokens. If demand exceeds capacity, transaction fees increase, requiring users to pay more tokens for their transactions.")
+			ux.Logger.PrintToUser(" ")
+			ux.Logger.PrintToUser(logging.Bold.Wrap("The blockchain's native token"))
+			ux.Logger.PrintToUser("Each blockchain on Avalanche has its own transaction fee token. To issue transactions users don't need to acquire ETH or AVAX and therefore the transaction fees are completely isolated.")
```
[the remark this answers is not in the document]
Comment (felipemadero) [reply pkg/vm/evmPrompts.go:284]: nop. this is not the case here. it is the subnet evm native token that any subnet evm has. for c-chain is avax, and so on.

**Pull request merged**